CorsixTH/Lua/world.lua

```diff
@@ -153,6 +153,9 @@ function World:World(app)
   for key, value in pairs(level_config.computer) do
     if value.Playing == 1 then
       self.hospitals[#self.hospitals + 1] = AIHospital(tonumber(key) + 1, self, avail_rooms)
+      if value.Name then
```

Comment:
Pass the name as the third parameter, just like the player hospital. Change the ... in aihospital to explicitly define the parameters the same as the parent class, only use the competitor table when name is nil, and remove this block.